### PR TITLE
feat(account): manage user consent from the self account service API

### PR DESCRIPTION
**Pull request by tcompiegne**
closes gravitee-io/issues#7680

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/services/impl/AccountServiceImpl.java [hunk 1]
```diff
@@ -185,6 +184,25 @@ public Single<Credential> getWebAuthnCredential(String id) {
                 });
     }
 
+    @Override
+    public Single<List<ScopeApproval>> getConsentList(User user, Client client) {
+        Flowable<ScopeApproval> source = (client != null) ?
+                scopeApprovalService.findByDomainAndUserAndClient(domain.getId(),  user.getId(), client.getClientId()) :
+                scopeApprovalService.findByDomainAndUser(domain.getId(), client.getId());
```
Comment (remisultan) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/services/impl/AccountServiceImpl.java:191]:
```suggestion
                scopeApprovalService.findByDomainAndUser(domain.getId(), user.getId());
```
Can use be null also here ?

Comment (remisultan) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/services/impl/AccountServiceImpl.java:191]:
```suggestion
                scopeApprovalService.findByDomainAndUser(domain.getId(), user.getId());
```
Also can user be null ? 

```suggestion
Maybe it's safer to go with

if (client != null){
  return scopeApprovalService.findByDomainAndUserAndClient(domain.getId(),  user.getId(), client.getClientId()).toList()
} else if (user != null){
  return scopeApprovalService.findByDomainAndUser(domain.getId(), user.getId()).toList()
}

return Single.just(List.of());
```

Comment (remisultan) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/services/impl/AccountServiceImpl.java:191]:
```suggestion
                scopeApprovalService.findByDomainAndUser(domain.getId(), user.getId());
```

Also can user be null ? 
Maybe it's safer to go with

```suggestion

if (client != null){
  return scopeApprovalService.findByDomainAndUserAndClient(domain.getId(),  user.getId(), client.getClientId()).toList()
} else if (user != null){
  return scopeApprovalService.findByDomainAndUser(domain.getId(), user.getId()).toList()
}

return Single.just(List.of());
```

**Review by remisultan — CHANGES_REQUESTED** [review]
How do we test the PR ?

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/services/impl/AccountServiceImpl.java (hunk 1 above)
Comment (tcompiegne) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/services/impl/AccountServiceImpl.java]: FP

File: docs/self-service-account-api-descriptor.yml [hunk 2]
```diff
@@ -339,6 +339,40 @@ paths:
...
+      produces:
+        - application/json
+      responses:
+        '200':
+          description: The user consent list
+          schema:
+            $ref: '#/definitions/UserConsentList'
+        '401':
+          description: Invalid Token
+  /api/consent/{consentId}:
```
Comment (leleueri) [inline docs/self-service-account-api-descriptor.yml:356]: Delete method is missing in the swagger.

File: docs/self-service-account-api-descriptor.yml [hunk 3]
```diff
@@ -339,6 +339,40 @@ paths:
...
+    get:
```
Comment (leleueri) [inline docs/self-service-account-api-descriptor.yml:357]: Don't see the interest to have a GET method on a specific consent here as the listContent already provide all the information.

**Review by leleueri — CHANGES_REQUESTED** [review]

File: docs/self-service-account-api-descriptor.yml (hunk 2 above)
Comment (tcompiegne) [reply docs/self-service-account-api-descriptor.yml:356]: FP

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/services/impl/AccountServiceImpl.java [hunk 4]
```diff
@@ -185,6 +184,25 @@ public Single<Credential> getWebAuthnCredential(String id) {
                 });
     }
 
+    @Override
+    public Single<List<ScopeApproval>> getConsentList(User user, Client client) {
+        Flowable<ScopeApproval> source = (client != null) ?
```
[the remark this answers is not in the document]
Comment (tcompiegne) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/services/impl/AccountServiceImpl.java:189]: FP

**Review by leleueri — APPROVED** [review]

**Pull request merged**